packages/roosterjs-plugin-picker/lib/PickerPlugin.ts

```diff
@@ -222,11 +225,16 @@ export default class EditorPickerPlugin implements EditorPickerPluginInterface {
                 this.handleKeyDownEvent(event);
             } else if (
                 this.dataProvider.shiftHighlight &&
-                (keyboardEvent.which == UP_ARROW_CHARCODE ||
-                    keyboardEvent.which == DOWN_ARROW_CHARCODE)
+                (this.pickerOptions.isHorizontal
+                    ? keyboardEvent.which == LEFT_ARROW_CHARCODE ||
+                      keyboardEvent.which == RIGHT_ARROW_CHARCODE
+                    : keyboardEvent.which == UP_ARROW_CHARCODE ||
+                      keyboardEvent.which == DOWN_ARROW_CHARCODE)
             ) {
                 this.dataProvider.shiftHighlight(
-                    keyboardEvent.which == DOWN_ARROW_CHARCODE ? true : false
+                    this.pickerOptions.isHorizontal
+                        ? keyboardEvent.which == RIGHT_ARROW_CHARCODE ? true : false

```

Comment:
Just use "keyboardEvent.which == RIGHT_ARROW_CHARCODE" is ok, sme for the next line